xdebug_handler_dbgp.c

```diff
@@ -1199,7 +1281,9 @@ DBGP_FUNC(source)
 	function_stack_entry *fse;
 
 	if (!CMD_OPTION_SET('f')) {
-		if ((fse = xdebug_get_stack_tail(TSRMLS_C))) {
+		if(XG(status) != DBGP_STATUS_BREAK&&XG(status) !=DBGP_STATUS_STARTING){
+			RETURN_RESULT(XG(status), XG(reason), XDEBUG_ERROR_COMMAND_UNAVAILABLE);
```

Comment:
What's the reason for doing this check not right away at the top-level?

Reply:
If IDE call without `f` flag, then the status should be `break` or `starting`.
If IDE call with `f` flag, the requirement is unnecessary.